docs/usage/administration/hardening.md

```diff
@@ -0,0 +1,52 @@
+# Hardening 
+**Disclaimer: This guide is just a collection of ideas to make a synapse homeserver more secure - It is far from complete!**
+
+## General hardening
+- SSH hardening (key-based login only, change port, etc.)
+- Basic firewall (e.g. via `iptables`) - The following configuration is necessary for a matrix server to work properly (**This setup is not bulletproof, but should be a quite secure default!**):
+	- incoming https rule for basic server functionality, e.g.
+	`iptables -A INPUT -p tcp --dport 443 -m conntrack --ctstate NEW -j ACCEPT`
+	- outgoing rule for https (the server has to "speak" to other homeservers), e.g.
+	`iptables -A OUTPUT -p tcp --dport 443 -m conntrack --ctstate NEW -j ACCEPT`
```

Comment:
there's no way to know for certain which ports a remote homeserver wants to listen on. Port 443 and 8448 are *common*, but for example `matrix.org` used 8443 until relatively recently.

In short, I think you'll need to make this rather broader.

Reply:
I think this boils down to the old security vs. functionality problem. In my case of a little friends-and-family homeserver, I had no issues (after adding the 8448 rule) yet. As soon as I encounter a connectivity problem with another homeserver, I review my firewall configuration and enable temporary logging of discarded packages. This way I am able to estimate the need for a new rule regarding a specific port.
But I understand that this is only my preferred way of doing things.. one could easily argue that less restrictive rules on the outbound chain impose only a minor security risk.
Regarding the guide, I'll try to make it clear that this is a more secure, less fail safe approach.

Reply:
I'm afraid if your server does this, it is not compliant with the matrix spec. It might work most of the time for you, but please don't recommend it for others.

In general, I don't really think you get much security by restricting outbound connections by port.

Reply:
So the general recommendation should not include any outbound restrictions, got you.
I'll update the guide accordingly!